File: core/src/main/java/com/pluxity/station/StationCodeRepository.java
package com.pluxity.station;

import java.util.List;
import org.springframework.data.jpa.repository.JpaRepository;

public interface StationCodeRepository extends JpaRepository<StationCode, Long> {
    List<StationCode> findByStationOrderByCreatedAtDesc(Station station);

    void deleteByStation(Station station);

    List<StationCode> findByStationInOrderByCreatedAtDesc(List<Station> stationIds);
}

File: core/src/main/java/com/pluxity/station/StationCodeRepository.kt
package com.pluxity.station

import org.springframework.data.jpa.repository.JpaRepository

interface StationCodeRepository : JpaRepository<StationCode, Long> {
    fun findByStationOrderByCreatedAtDesc(station: Station): List<StationCode>

    fun deleteByStation(station: Station)

    fun findByStationInOrderByCreatedAtDesc(stationIds: List<Station>): List<StationCode>
}
